app/jobs/base.rb

```diff
@@ -461,6 +461,45 @@ def self.enqueue_at(datetime, job_name, opts = {})
     enqueue_in(secs, job_name, opts)
   end
 
+  @@enqueue_after = []
```

Comment:
I don't think we should hold the callback in memory because a process restart will cause the callback to be lost.

Reply:
Redis seems like a good storage option here.

Reply:
Do we also need to think about timeouts here to account for the case where jobs don't complete within the given timeframe? We don't want the callback to linger around forever.